src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -147,13 +147,21 @@ internal async Task InitializeOnActivatedAsync(CancellationToken cancellationTok
                 _packageSources.CurrentChanged += OnSelectedPackageSourceChanged;
                 PackageSourcesListBox.GotFocus += PackageSourcesListBox_GotFocus;
                 PackageSourcesListBox.DataSource = _packageSources;
+                for (int i = 0; i < packageSources.Count; i++)
+                {
+                    PackageSourcesListBox.SetItemChecked(i, packageSources[i].IsEnabled);
```

Comment:
>PackageSourcesListBox.SetItemChecked(i, packageSources[i].IsEnabled); [](http://example.com/codeflow?start=20&length=69)

Doesn't this assume that the order in which different sources appear in the UI and allPackageSources are the same? Can we really assume that?

Reply:
I think this is just iterating the collection, so order isn't important.
It's just making sure the config value matches the checkbox.

Reply:
This is using `packageSources`, not `allPackageSources`. Up on line 143, it does `var packageSources = allPackageSources.Where(ps => !ps.IsMachineWide).ToList();`. And then the `BindingSource` is also initialized with `packageSources.ToList()` on line 146. The same order is guaranteed by `ToList`, so I feel like the same order can be assumed, yes.

> I think this is just iterating the collection, so order isn't important.

If not all sources have the same enabled state, then order is important, since you don't want to to set the wrong checkbox state for an item. But as per above, I believe the order can be assumed to be the same.